sklearn/preprocessing/tests/test_data.py

```diff
@@ -1535,9 +1528,17 @@ def _check_one_hot(X, X2, cat, n_features):
     assert_array_equal(toarray(B), toarray(D))
 
 
+def test_one_hot_encoder_string():
+    X = [['cat', 'domestic'], ['wolf', 'wild']]
+    enc = OneHotEncoder()
+    enc.fit(X)
+    Xtr = enc.transform([['cat', 'wild']])
+    assert_array_equal(toarray(Xtr), [[1, 0, 0, 1]])
+
+
 def test_one_hot_encoder_categorical_features():
     X = np.array([[3, 2, 1], [0, 1, 1]])
-    X2 = np.array([[1, 1, 1]])
+    X2 = np.array([[3, 1, 1]])
```

Comment:
Since we decided to rename `n_values` to `values` should I leave the default `n_values` as `None` ? The default for `values` will be `"seen"`. `values` can be set to `"auto"`, `"seen"`, ints, lists of ints or list of lists. 

And as currently implemented if `n_values` is not `None`, we will issue a warning and set `values = n_values`